packages/app/src/runner/event-manager.ts

```diff
@@ -569,8 +581,13 @@ export class EventManager {
       this.localBus.emit('script:error', err)
     })
 
-    Cypress.on('test:before:run:async', (_attr, test) => {
+    Cypress.on('test:before:run:async', async (_attr, test) => {
       this.studioStore.interceptTest(test)
+
+      // if the experimental flag is on and we are in a chromium based browser, determine if garbage collection is needed
+      if (Cypress.config('experimentalMemoryManagement') && Cypress.isBrowser({ family: 'chromium' })) {
+        await Cypress.backend('maybe:collect:garbage', { test: { title: test.title, order: test.order, currentRetry: test.currentRetry() } })
```

Comment:
How about `request:collect:garbage`, because requests can be denied

```suggestion
        await Cypress.backend('request:collect:garbage', { test: { title: test.title, order: test.order, currentRetry: test.currentRetry() } })
```

or focus less on the outcome? `check:memory:pressure` and if that pressure happens to be high also GC?
```suggestion
        await Cypress.backend('check:memory:pressure', { test: { title: test.title, order: test.order, currentRetry: test.currentRetry() } })
```

🤷‍♂️ , i dunno just not a fan of maybe statements.

Reply:
I'm fine with using `maybe` as it matches other functions within the codebase but if others want it changed, I'm fine with updating it.

Reply:
renamed to `check:memory:pressure` [9e7ee14](https://github.com/cypress-io/cypress/pull/25462/commits/9e7ee14ed4b032d6dec6fc6f3e08b475de57ef32)